### DISCUSSION TITLE
Dispatch success action for RTKQ endpoint in tests

### DISCUSSION THREAD

**Discussion by dumaron** [reduxjs/redux-toolkit#4608] [Q&A] [answered]
I'm struggling to write a unit test for a slice that listens to successful actions coming from a RTK-Query API slice. Instead of initializing a whole store with the two slices and then using the `.initiate()` function, I'd prefer to just dispatch on the barebone reducer the success action and then look for the result.

A working test I have for my reducer is this:
```typescript
it('should have 0 value for my value after initialization', () => {
    const state = mySlice.reducer(initialState, { type: 'init' })
    expect(state.myValue).toBe(0)
})
```

When I then try to set a similar test for a RTKQ success action, I don't know how to determine the right type for the action:
```typescript
it('should update my value when it has been persisted on server', () => {
    const state = ebaSlice.reducer(
      initialState,
      myApi.endpoints.persistValue.???, // <- maybe an action creator, or a `type`
    )
    expect(state.myValue).toBe(1)
})
```

The slice is hooked to the API slice in the classic way:
```typescript
builder.addMatcher(myApi.endpoints.persistValue.matchFulfilled, (state, { payload }) => {
       state.myValue = payload.value
    }
})
```

Is there anything I can use to dispatch this successful action directly on the reducer?
Thanks!

versions:
- @reduxjs/toolkit: `1.9.1`
- redux: `4.2.0`

**Comment by markerikson** [comment, marked as answer]
The type is going to be `"myApi/execute/fulfilled"`, but it's made more specific with the endpoint name and request ID.

Generally the right answer here _is_ to create a real store with the actual API slice and use the thunks as it would normally work.
